src/components/Checkout/CheckoutDropInPaymentWrapper.vue

```diff
@@ -168,7 +172,22 @@ export default {
 		},
 		validateGuestBasketAndCheckout() {
 			this.$emit('updating-totals', true);
-			this.validateGuestBasket(this.email, this.emailUpdates)
+			// Set the default checkout validation method
+			let validationMethod = this.validateGuestBasket;
+			const validationPayload = {
+				email: this.email,
+				emailUpdates: this.emailUpdates,
+			};
+			// If promo guest checkout is enabled, use the promo guest checkout validation method.
+			// This method validates the lender email for promo first before running the guest checkout method
+			// in checkout utils.
+			if (promoGuestCheckoutEnabled) {
```

Comment:
need to set this as a prop here and pass it in from the parent.

Reply:
Oh or just use the `isGuestCheckout` which I think you're already passing this value to!

Reply:
This is just a draft for you to see the direction I'm taking this in, but I know the code was not in a state where it was functioning perfectly quite yet.

I've since added the prop and done a bunch of other things, still just trying to ensure that the guest checkout is working while features are being gated by this flag.  Credit card payment seems to be going through just fine, but Kiva Credit payments are having trouble at the moment.